core/src/types/operator/blocking_operator.rs

```diff
@@ -680,6 +680,98 @@ impl BlockingOperator {
         Ok(())
     }
 
+    ///
+    /// # Notes
+    ///
+    /// We don't support batch delete now.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use anyhow::Result;
+    /// # use futures::io;
+    /// # use opendal::BlockingOperator;
+    /// # fn test(op: BlockingOperator) -> Result<()> {
+    /// op.remove(vec!["abc".to_string(), "def".to_string()])?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn remove(&self, _paths: Vec<String>) -> Result<()> {
+        Ok(())
+    }
+
+    /// remove will remove files via the given paths.
+    ///
+    /// remove_via will remove files via the given stream.
+    ///
+    /// We will delete by chunks with given batch limit on the stream.
+    ///
+    /// # Notes
+    ///
+    /// We don't support batch delete now.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use anyhow::Result;
+    /// # use futures::io;
+    /// # use opendal::BlockingOperator;
+    /// # fn test(op: BlockingOperator) -> Result<()> {
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn remove_via(&self) -> Result<()> {
+        //     let stream = stream::iter(vec!["abc".to_string(), "def".to_string()]);
+        // op.remove_via(stream)?;
+        Ok(())
+    }
+
+    /// Remove the path and all nested dirs and files recursively.
+    ///
+    /// # Notes
+    ///
+    /// We don't support batch delete now.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use anyhow::Result;
+    /// # use futures::io;
+    /// # use opendal::BlockingOperator;
+    /// # fn test(op: BlockingOperator) -> Result<()> {
+    /// op.remove_all("path/to/dir")?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn remove_all(&self, path: &str) -> Result<()> {
+        let meta = match self.stat(path) {
+            Ok(metadata) => metadata,
+
+            Err(e) if e.kind() == ErrorKind::NotFound => return Ok(()),
+
+            Err(e) => return Err(e),
+        };
+
+        if meta.mode() != EntryMode::DIR {
+            return self.delete(path);
+        }
+
+        let obs = self.scan(path)?;
+
+        obs.for_each(|v| {
+            if (match v {
+                Ok(entry) => self.inner().blocking_delete(entry.path(), OpDelete::new()),
+                Err(e) => Err(e),
```

Comment:
We need to return this error.

Reply:
hi, may I ask how I should debug this?

Reply:
I prefer to expand as a `for` instead for `for_each`.

Reply:
> hi, may I ask how I should debug this? I don't find this in documents.

You can try `cargo test fs`